libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/extensions/WaypointEx.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.navigation.base.internal.extensions
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.internal.route.Waypoint
+import com.mapbox.navigation.base.trip.model.RouteProgress
+
+/**
+ * Return waypoints that are requested explicitly
+ */
+fun List<Waypoint>.requestedWaypoints(): List<Waypoint> =
+    this.filter {
+        when (it.type) {
+            Waypoint.REGULAR,
+            Waypoint.SILENT -> true
+            Waypoint.EV_CHARGING -> false
+            else -> false
```

Comment:
the test does not guarantee that every place where it is used be updated along with new waypoint types, also it limits adding new fields to the companion object. 

let me know what you think: I added internal enum and keep `Def` to make it forward compatible (also, it's used by other modules) https://github.com/mapbox/mapbox-navigation-android/pull/6005/commits/0bd7ccf365b1b9ec8a09e4aabeed5aacd12efee3

Reply:
It's not guaranteed that we'll use internalType, not type everywhere in our code. It's kind of a compromise though.

> the test does not guarantee that every place where it is used be updated along with new waypoint types, also it limits adding new fields to the companion object.

It was out of context perhaps, I suggested adding the parameterized test for `getRequestedWaypoints` and so on and check the input values. So that when you add a new value, the test will fail because you haven't tested this value.You'll look at the test and go on and fix the original method.

> also it limits adding new fields to the companion object

You can always add `+1` with a comment that it's for this field. I don't think there would be too many false positives especially considering the fact that it's easily fixed.

I'll leave the choice to you, both options would work fine for me.

Reply:
discussed: approaches are similar, keep using enums